projects/lib/src/board/sittuyinboard.cpp

```diff
@@ -0,0 +1,340 @@
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    Cute Chess is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with Cute Chess.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include "sittuyinboard.h"
+
+namespace Chess {
+
+SittuyinBoard::SittuyinBoard()
+	: MakrukBoard(),
+	  m_inSetUp(true)
+{
+	// Movements as in makruk, the Elephant has Silver General Movement
+	setPieceType(Pawn, tr("pawn"), "P");                      //!< Ne: Feudal Lord P
+	setPieceType(Knight, tr("knight"), "N", KnightMovement);  //!< Myin: Horse N
+	setPieceType(Elephant, tr("elephant"), "S", SilverGeneralMovement, "E"); //! Sin: Elephant E
+	setPieceType(Rook, tr("rook"), "R", RookMovement);        //! Yahhta: Chariot R
+	setPieceType(General, tr("general"), "F", FerzMovement);  //! Sit-ke: General G
+	setPieceType(King, tr("king"), "K");                      //! Min-gyi: King K
+}
+
+Board* SittuyinBoard::copy() const
+{
+	return new SittuyinBoard(*this);
+}
+
+QString SittuyinBoard::variant() const
+{
+	return "sittuyin";
+}
+
+QString SittuyinBoard::defaultFenString() const
+{
+	return "8/8/4pppp/pppp4/4PPPP/PPPP4/8/8[KFRRSSNNkfrrssnn] w - 0 0 1";
+	//Federation notation, ref. Maung Maung Lwin
+	//return "8/8/4pppp/pppp4/4PPPP/PPPP4/8/8[KGRREENNkgrreenn] w - 0 0 1";
+}
+
+bool SittuyinBoard::variantHasDrops() const
+{
+	return true;
+}
+
+bool SittuyinBoard::variantHasOptionalPromotions() const
+{
+	return true;
+}
+
+QList< Piece > SittuyinBoard::reservePieceTypes() const
+{
+	QList< Piece > types;
+	for (Side side: { Side::White, Side::Black} )
+	{
+		for (int type = Knight; type <= King; type++)
+			types << Piece(side, type);
+	}
+	return types;
+}
+
+MakrukBoard::CountingRules SittuyinBoard::countingRules() const
+{
+	return BareKing;
+}
+
+bool SittuyinBoard::kingsCountAssertion(int whiteKings, int blackKings) const
+{
+	Piece whiteKing(Side::White, King);
+	Piece blackKing(Side::Black, King);
+	return whiteKings + reserveCount(whiteKing) == 1 && blackKings + reserveCount(blackKing) == 1;
+}
+
+int SittuyinBoard::promotionRank(int file) const
+{
+	return std::max(file , height() - 1 - file);
+}
+
+void SittuyinBoard::addPromotions(int sourceSquare, int targetSquare, QVarLengthArray< Move >& moves) const
+{
+	moves.append(Move(sourceSquare, targetSquare, General));
+}
+
+void SittuyinBoard::generatePawnMoves(int square,
+				     QVarLengthArray<Move>& moves) const
+{
+	// Generate moves for Pawn (Feudal Lord)
+	QVarLengthArray< Move > moves1;
+	ShatranjBoard::generateMovesForPiece(moves1, Pawn, square);
+
+	// Add normal moves and demote shatranj promotions
+	for (const Move m: moves1)
```

Comment:
You can use a const reference to avoid copying:

`for (const Move& m : moves1)`

Reply:
Changed this accordingly, thank you!